internal/translation/deployment/compare.go

```diff
@@ -298,18 +298,23 @@ func diskAutoscalingConfigAreEqual(desired, current *akov2.DiskGB) bool {
 }
 
 func computeAutoscalingConfigAreEqual(desired, current *akov2.ComputeSpec) bool {
-	if desired == nil && current == nil {
-		return true
+	if desired == nil {
+		desired = &akov2.ComputeSpec{}
 	}
 
-	if (desired != nil && current == nil) || (desired == nil && current != nil) {
-		return false
+	if current == nil {
+		current = &akov2.ComputeSpec{}
 	}
 
 	if desired.Enabled != nil && !areEqual(desired.Enabled, current.Enabled) {
 		return false
 	}
 
+	// If autoscaling is disabled rest of fields don't matter.
```

Comment:
I believe we are missing cases here. This code breaks 5 unit test cases. In general if autoscale is expected to be disabled AND it is indeed disabled in Atlas, the rest of the fields do not matter, BUT if the Atlas side is enabled, then  we cannot return true here.